charts/tfy-agent/templates/tfy-agent-proxy-clusterrole-ns.yaml

```diff
@@ -70,13 +70,24 @@ rules:
     resources: ["notebooks"]
     verbs: ["get", "update"]
 
+  {{/* This allows us to create intercepts for redirection / shadowing traffic */}}
   - apiGroups: ["networking.istio.io"]
     resources: ["envoyfilters"]
     verbs: ["create", "patch", "delete"]
 
   - apiGroups: ["flyte.lyft.com"]
     resources: ["flyteworkflows"]
     verbs: ["create", "patch", "get", "delete"]
+
+  {{/* This allows us to apply VirtualService for Spark UI when applying a SparkApplication (Spark Job Run) */}}
+  - apiGroups: ["networking.istio.io"]
+    resources: ["virtualservices"]
+    verbs: ["create", "get", "patch"]
+
+  {{/* This allows us to apply/delete SparkApplication (Spark Job Run) */}}
+  - apiGroups: ["sparkoperator.k8s.io"]
+    resources: ["sparkapplications"]
+    verbs: ["create", "get", "delete"]
```

Comment:
Just for my understanding, why do we need `get`?

Reply:
I'll remove it for now, I was thinking we might need it we want to get the driver pod uid from status field and patch the VS